### DISCUSSION TITLE
How to create a listener, like in TypeORM

### DISCUSSION THREAD

**Discussion by georgekrax** [prisma/prisma#10595] [Q&A] [answered]
Hello everyone,

Is there a way to create a listener in Prisma, like in TypeORM? Also, is there any guide on how to migrate from TypeORM to Prisma? With the OOP architecture that TypeORM uses, I can place reusable functions for a specific model (Entity) and use them anywhere. Is there a way to do something similar in Prisma? Or, do I have to create separate functions that take as a parameter the Prisma client object, and import them every time I want to use them?

Any help or advice is appreciated!

**Comment by austincrim** [comment, marked as answer]
Hey @georgekrax 👋!

Prisma does not have a listener concept that maps directly to TypeORM listeners. Our general advice would be to create small, reusable functions if you find yourself repeating the same query or operation a lot.

We have [this migration guide](https://www.prisma.io/docs/guides/migrate-to-prisma/migrate-from-typeorm) on our docs that discusses some basic scenarios! Let me know if I can answer any more specific questions, though.

**Reply by georgekrax** [reply]
Thank you for your helpful response!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
